### PR TITLE
Catalog image lazy load

### LINKED ISSUE #27032: Add image lazy loading
### Description (*)

In many situations, catalog lists can contain many images, while most of them remain under the first fold. 

When running an audit using Lighthouse, it tells us the following.
image

So we need to add the `loading="lazy"` attribute to the images on the catalog list. This attribute is supported on the following browsers ([source](https://caniuse.com/#feat=loading-lazy-attr)):
- [Chrome >= 76](https://blog.chromium.org/2019/10/automatically-lazy-loading-offscreen.html)
  - Edge >= 79
  - Opera >= 64
  - Android Browser >= 80
  - Chrome for Android >= 80
- [Firefox >= 75](https://www.ghacks.net/2020/02/15/firefox-75-gets-lazy-loading-support-for-images/) 

In Chromium's aforementioned blog post, it seems that lazy loading is enabled for all websites on Chrome Android browsers with Lite Mode enabled.

### Expected behavior (*)

Better performance for catalog pages. This is achieved by adding the `loading="lazy"` attribute to the `image_with_borders.phtml` template. 

### Benefits

Better performance for both users and Lighthouse reports.

### Additional information

Not for now

### PR DESCRIPTION
**Pull request by tdgroot**
### Description (*)
This changeset adds lazy loading to the catalog product list.

### Fixed Issues (if relevant)
1. magento/magento2#27032: Add image lazy loading

### Manual testing scenarios (*)
1. Have a product list with more than 16 products, all having unique products.
2. Check inspector if images are loaded through scrolling.

### Related pull requests
- #27034: FIX: wrong img size attributes (this PR includes those changes)

### Questions or comments
I think discussions on the topic 'image lazy loading' should be held on #27032. 

Any code reviewing comments can be placed in this PR.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml [hunk 1]
```diff
@@ -4,16 +4,20 @@
...
-            <?= $block->escapeHtmlAttr($block->getCustomAttributes()) ?>
-            src="<?= $block->escapeUrl($block->getImageUrl()) ?>"
-            max-width="<?= $block->escapeHtmlAttr($block->getWidth()) ?>"
-            max-height="<?= $block->escapeHtmlAttr($block->getHeight()) ?>"
-            alt="<?= /* @noEscape */ $block->stripTags($block->getLabel(), null, true) ?>"/></span>
+        <img class="<?= $escaper->escapeHtmlAttr($block->getClass()) ?>"
+             <?= $escaper->escapeHtmlAttr($block->getCustomAttributes()) ?>
+             src="<?= $escaper->escapeUrl($block->getImageUrl()) ?>"
+             loading="lazy"
+             width="<?= $escaper->escapeHtmlAttr($block->getWidth()) ?>"
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml]: Could you explain why you changed max-width and max-height to just width and height? It looks not related to this PR.

As max width and max height are not valid attributes - seems like we could just remove them

File: app/code/Magento/Catalog/view/frontend/templates/product/image.phtml [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
-    width="<?= $block->escapeHtmlAttr($block->getWidth()) ?>"
-    height="<?= $block->escapeHtmlAttr($block->getHeight()) ?>"
-    alt="<?= /* @noEscape */ $block->stripTags($block->getLabel(), null, true) ?>" />
+<img class="photo image <?= $escaper->escapeHtmlAttr($block->getClass()) ?>"
+     <?= $escaper->escapeHtml($block->getCustomAttributes()) ?>
+     src="<?= $escaper->escapeUrl($block->getImageUrl()) ?>"
+     loading="lazy"
+     width="<?= $escaper->escapeHtmlAttr($block->getWidth()) ?>"
+     height="<?= $escaper->escapeHtmlAttr($block->getHeight()) ?>"
+     alt="<?= /* @noEscape */ $block->stripTags($block->getLabel(), null, true) ?>" />
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/view/frontend/templates/product/image.phtml]: Looks like this value still need to be escaped, could you fix that?

File: app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml [hunk 3]
```diff
@@ -4,16 +4,20 @@
...
+             height="<?= $escaper->escapeHtmlAttr($block->getHeight()) ?>"
+             alt="<?= /* @noEscape */ $block->stripTags($block->getLabel(), null, true) ?>"/></span>
```
Comment (ihor-sviziev) [inline app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml]: Looks like this value still need to be escaped, could you fix that?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml (hunk 1 above)
Comment (tdgroot) [reply app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml]: The width and height attributes are needed to preserve the aspect ratio of the image when it is not loaded yet.

Comment (ihor-sviziev) [reply app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml]: Ok, no problem, but need just test it carefully on different devices, because it could do regression

**Comment by ihor-sviziev** [comment]
@magento run  Functional Tests B2B

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-7006](https://jira.corp.magento.com/browse/ENGCOM-7006) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed
